DataProc/CPass0/mergeMakeOCDB.byComponent.perStage.sh

```diff
@@ -472,7 +501,11 @@ copyScripts()
 
 extractFileNamesFromXMLCollection()
 {
-    grep turl|sed 's|^.*turl\s*=\s*"\s*\([a-zA-Z]*://.*\.root\).*$|\1|g'
+    if [[ ! $ROOTSYS || ! $ALIDPG_ROOT ]]; then
```

Comment:
This part _does_ affect Grid jobs. As a matter of fact, we've replaced a, er, "homemade" XML parser with a proper one, see the new `readAliEnCollection.sh` script. The reason for this replacement is that it misbehaved for generic URLs.

**We have tested it successfully:** the parser is used if ROOT and AliDPG are both available in the job's environment (which should always be the case), otherwise we fall back to the old method.